File: octopoints_flutter/lib/service/model/rule.dart
class Rule {
  final int id;
  final int matchId;
  int winners;
  int total;
  bool battleRoyale;

  Rule({
    this.id = 0,
    required this.matchId,
    required this.winners,
    required this.total,
    required this.battleRoyale,
  });

  Rule setTotal(int total) {
    this.total = total;
    return this;
  }

  Rule setWinners(int winners) {
    this.winners = winners;
    return this;
  }
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/model/Rule.kt
package dev.tpavani.octopoints.service.model

class Rule(val matchId: Int, var winners: Int, var total: Int, var battleRoyale: Boolean) {
    private var id: Int? = null

    constructor(id: Int, matchId: Int, winners: Int, total: Int, battleRoyale: Boolean):this(matchId, winners, total, battleRoyale){
        this.id = id
    }

    fun getId(): Int{
        if(this.id == null){
            return 0
        }
        return this.id!!
    }

    fun setId(id: Int): Rule{
        this.id = id
        return this
    }
}